hack: split up test-dockerized.sh

pohly opened the pull request:
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

This enables testing command and integration tests separately in CI jobs. The goal is to reduce pull-kubernetes-integration to testing really just the Go test/integration tests and to add a pull-kubernetes-cmd job for command tests.

hack/jenkins/test-dockerized.sh does the same as before because some jobs will probably continue to use it.

#### Which issue(s) this PR fixes:

Discussed in https://kubernetes.slack.com/archives/C09QZ4DQB/p1740083887716479

#### Does this PR introduce a user-facing change?
```release-note
NONE
```

/hold

For testing with to-be-defined canary presubmits.

/cc @dims @aojea @BenTheElder 
/sig testing

pohly committed:
hack: split up test-dockerized.sh

This enables testing command and integration tests separately in CI jobs. The
goal is to reduce pull-kubernetes-integration to testing really just the Go
test/integration tests and to add a pull-kubernetes-cmd job for command tests.

hack/jenkins/test-dockerized.sh does the same as before because some jobs will
probably continue to use it.

dims commented:
/approve
/lgtm

aojea commented:
shouldn't be better to prefix the new files with test, as test-cmd-dockerized and test-integration-dockerized ?

pohly commented:
> shouldn't be better to prefix the new files with test, as test-cmd-dockerized and test-integration-dockerized ?

That would be inconsistent with `hack/jenkins/verify-dockerized.sh`. :shrug: 

/test pull-kubernetes-verify-canary
/test pull-kubernetes-cmd-canary

pohly commented:
/test pull-kubernetes-integration-canary

pohly commented:
/test pull-kubernetes-cmd-canary

I've simplified cmd-dockerized.sh: it was setting variables that had no effect for it.

pohly commented:
/test pull-kubernetes-integration-canary

pohly commented:
@aojea: the canary jobs worked. Okay to move ahead or shall I rename the scripts?

pohly commented:
> shouldn't be better to prefix the new files with test, as test-cmd-dockerized and test-integration-dockerized ?

I've changed my mind and now agree with this suggestion, so I renamed. The pattern then is `jenkins/<make target>-dockerized.sh`, not `jenkins/<kubernetes job suffix>-dockerized.sh` as it was before.

@aojea: LGTM?

aojea commented:
/lgtm
/approve

Thanks

pohly commented:
/hold cancel
